lib/modules/dosomething/dosomething_reportback/dosomething_reportback.query.inc

```diff
@@ -123,25 +121,26 @@ function dosomething_reportback_get_reportbacks_query($params) {
   return $result;
 }
 
-
 /**
- * Filter out requested flagged Reportbacks based on user permissions.
- *
- * @param  array  $params
- * @return mixed
+ * Determine if user has access to flagged Reportbacks based on user permission. 
+ * @param array $results
  */
-function dosomething_reportback_filter_flagged_reportbacks($params) {
+function dosomething_reportback_accessible_results ($results) {
   if (user_access('view any reportback')) {
-    if (isset($params['flagged'])) {
-      unset($params['flagged']);
+    return $results;
+  }
+  else {
+    foreach ($results as $key => $reportback) {
+      if ($reportback->flagged == 1) {
```

Comment:
ooo, I have a function for that... [`dosomething_helpers_convert_string_to_boolean()`](https://github.com/DoSomething/phoenix/blob/f0caf06c5ccacc05c9fceb659d1b51736186de83/lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module#L127). Returns a _true_ boolean value from a string :)